commands/kubernetes.go

```diff
@@ -579,6 +711,121 @@ func buildNodePoolUpdateRequestFromArgs(c *CmdConfig, r *godo.KubernetesNodePool
+	return clientcmd.ModifyConfig(kubectlDefaults, *currentConfig, false)
+}
+
+func removeFromKubeconfig(kubeconfig []byte) error {
+	remote, err := clientcmd.Load(kubeconfig)
+	if err != nil {
+		return err
+	}
+	kubectlDefaults := clientcmd.NewDefaultPathOptions()
+	currentConfig, err := kubectlDefaults.GetStartingConfig()
+	if err != nil {
+		return err
+	}
+	if err := removeKubeconfig(remote, currentConfig); err != nil {
+		return fmt.Errorf("couldn't use the kubeconfig info received, %v", err)
+	}
+	return clientcmd.ModifyConfig(kubectlDefaults, *currentConfig, false)
+}
+
+// mergeKubeconfig merges a remote cluster's config file with a local config file,
+// assuming that the current context in the remote config file points to the
+// cluster details to add to the local config.
+func mergeKubeconfig(remote, local *clientcmdapi.Config) error {
+	remoteCtx, ok := remote.Contexts[remote.CurrentContext]
+	if !ok {
+		// this is a bug in the backend, we received incomplete/non-sensical data
+		return fmt.Errorf("the remote config has no context entry named %q. This is a bug, please open a ticket with DigitalOcean!",
+			remote.CurrentContext,
+		)
+	}
+	remoteCluster, ok := remote.Clusters[remoteCtx.Cluster]
+	if !ok {
+		// this is a bug in the backend, we received incomplete/non-sensical data
+		return fmt.Errorf("the remote config has no cluster entry named %q. This is a bug, please open a ticket with DigitalOcean!",
+			remoteCtx.Cluster,
+		)
+	}
+	remoteAuthInfo, ok := remote.AuthInfos[remoteCtx.AuthInfo]
+	if !ok {
+		// this is a bug in the backend, we received incomplete/non-sensical data
+		return fmt.Errorf("the remote config has no user entry named %q. This is a bug, please open a ticket with DigitalOcean!",
+			remoteCtx.AuthInfo,
+		)
+	}
+
+	local.Contexts[remote.CurrentContext] = remoteCtx
+	local.Clusters[remoteCtx.Cluster] = remoteCluster
+	local.AuthInfos[remoteCtx.AuthInfo] = remoteAuthInfo
+	return nil
+}
+
+// removeKubeconfig removes a remote cluster's config file from a local config file,
+// assuming that the current context in the remote config file points to the
+// cluster details to reomve from the local config.
+func removeKubeconfig(remote, local *clientcmdapi.Config) error {
+	remoteCtx, ok := remote.Contexts[remote.CurrentContext]
+	if !ok {
+		// this is a bug in the backend, we received incomplete/non-sensical data
+		return fmt.Errorf("the remote config has no context entry named %q. This is a bug, please open a ticket with DigitalOcean!",
+			remote.CurrentContext,
+		)
+	}
+
+	delete(local.Contexts, remote.CurrentContext)
+	delete(local.Clusters, remoteCtx.Cluster)
+	delete(local.AuthInfos, remoteCtx.AuthInfo)
+	if local.CurrentContext == remote.CurrentContext {
+		local.CurrentContext = ""
+	}
+	return nil
+}
+
+// waitForClusterRunning waits for a cluster to be running.
+func waitForClusterRunning(kube do.KubernetesService, clusterID string) error {
+	failCount := 0
```

Comment:
Printing out that the command is doing something here would be good. Right now it's just not showing anything while it's waiting